modules/tensor_mechanics/src/materials/PowerLawCreepStressUpdate.C

```diff
@@ -79,8 +49,9 @@ PowerLawCreepStressUpdate::computeStressInitialize(const Real /*effective_trial_
   if (_has_temp)
     _exponential = std::exp(-_activation_energy / (_gas_constant * _temperature[_qp]));
   else
-    _exponential = 1;
+    _exponential = 1.0;
 
+  mooseAssert(_t >= _start_time, "Time must be greater than start time");
   _exp_time = std::pow(_t - _start_time, _m_exponent);
 
   _creep_strain[_qp] = _creep_strain_old[_qp];
```

Comment:
@bwspenc Is this step necessary? If it is generic, it should be moved to the base class.

Reply:
I don't think it is necessary.

Reply:
Ah, I see. Testing it out, it's necessary if you do `_creep_strain[_qp] += plasticStrainIncrement;`. I'll take out both of these.

Reply:
If you don't set the current material property it will retain the value from the previous element. I just ran into the same thing in some code I'm working on. Just be careful about this because there will not be a runtime error and you will only notice this when looking at your results very carefully (you'll see an odd horizontal streaking in your properties).

Reply:
Huh. Do you mean it'll retain the value from the previous qp altogether? What is the initial value of a material property before being calculated? Where does `propagateQpStatefulPropertiesRadialReturn` fit in?

Reply:
Ah, I think my case was slightly different as I was setting the value of a non-stateful property using the data from a stateful property (hence the streaking). Here teh error would be even more subtle as the current material data would be cycled around and would probably hold the previous _older_ data. In any case it its necessary to write to the current data in _all_ cases. I haven't followed the code flow here so take it with a grain of salt, but you cannot do any of teh following

* not write to a property data entry (stateful or not does not matter)
* read before you write (e.g. `+=` operator)

Reply:
@tophmatthews If you do what you proposed, it will be handled correctly in all cases. Look at the models in BISON -- they do it the way you proposed.

Reply:
Yeah, that's what I thought. I don't think creep_strain is used before it is set.